state/execution_test.go

```diff
@@ -27,6 +33,59 @@ var (
 	nTxsPerBlock        = 10
 )
 
+func TestCreateProposalBlock_MempoolRemoveTxs(t *testing.T) {
+	var invalidTxs = [][]byte{{0x01}, {0x02}}
+	mockProxyApp := &mocks.AppConnConsensus{}
+
+	config := cfg.ResetTestRoot("node_create_proposal")
+	defer os.RemoveAll(config.RootDir)
+	app := &testApp{}
+	cc := proxy.NewLocalClientCreator(app)
+	proxyApp := proxy.NewAppConns(cc)
+	err := proxyApp.Start()
+	require.Nil(t, err)
+	defer proxyApp.Stop() //nolint:errcheck // ignore for tests
+
+	logger := log.TestingLogger()
+	state, stateDB, _ := makeState(1, 1)
+	proposerAddr, _ := state.Validators.GetByIndex(0)
+
+	// Make Mempool
+	mempool := mempl.NewCListMempool(
+		config.Mempool,
+		proxyApp.Mempool(),
+		state.LastBlockHeight,
+	)
+	mempool.SetLogger(logger)
+
+	txs := []types.Tx{[]byte{0x01}, []byte{0x02}, []byte{0x03}}
+	for i, tx := range txs {
+		err := mempool.CheckTx(tx, nil, mempl.TxInfo{})
+		assert.NoError(t, err)
+		assert.EqualValues(t, i+1, mempool.TxsBytes())
+	}
+
+	blockExec := sm.NewBlockExecutor(
+		stateDB,
+		logger,
+		mockProxyApp,
+		mempool,
+		sm.MockEvidencePool{},
+	)
+	mockProxyApp.On("CreateBlockSync", tmock.Anything, tmock.Anything).Return(&abcix.ResponseCreateBlock{
+		Txs:        [][]byte{{0x03}},
+		InvalidTxs: invalidTxs,
+	}, nil)
+
+	commit := types.NewCommit(0, 0, types.BlockID{}, nil)
+	blockExec.CreateProposalBlock(
+		1,
+		state, commit,
+		proposerAddr,
+	)
+	assert.EqualValues(t, 1, mempool.TxsBytes())
```

Comment:
can also call `GetNextTxBytes` and makes sure returning desired tx and `nil` starter